src/Featurizers/NormalizeFeaturizer.h

```diff
@@ -0,0 +1,325 @@
+
+        assert(data.Type == NormT);
+        return typename BaseType::TransformerUniquePtr(new NormalizeTransformer<InputT>(data.Norms));
+    }
+};
+
+} // namespace Details
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         NormalizeEstimator
+///  \brief         Chains up VectorNormsEstimator and NormalizeTransformerImpl
+///                 to normalize input vector data so that it's norm becoming 1
+///
+template <
+    typename InputT,
+    NormType NormT,
+    size_t MaxNumTrainingItemsV=std::numeric_limits<size_t>::max()
+>
+class NormalizeEstimator :
+    public Components::PipelineExecutionEstimatorImpl<
+        Components::VectorNormsEstimator<InputT, NormT, MaxNumTrainingItemsV>,
+        Details::NormalizeEstimatorImpl<InputT, NormT, MaxNumTrainingItemsV>
+    > {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType =
+        Components::PipelineExecutionEstimatorImpl<
+            Components::VectorNormsEstimator<InputT, NormT, MaxNumTrainingItemsV>,
+            Details::NormalizeEstimatorImpl<InputT, NormT, MaxNumTrainingItemsV>
+        >;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    NormalizeEstimator(AnnotationMapsPtr pAllColumnAnnotations, size_t colIndex);
+    ~NormalizeEstimator(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(NormalizeEstimator);
+};
+
+
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// |
+// |  Implementation
+// |
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+
+// ----------------------------------------------------------------------
+// |
+// |  NormalizeTransformer
+// |
+// ----------------------------------------------------------------------
+template <typename InputT>
+NormalizeTransformer<InputT>::NormalizeTransformer(std::vector<std::double_t> norms) :
+    _norms(std::move(norms)),
+    _row(0) {
+        for (auto it = norms.begin(); it != norms.end(); ++it) {
+            if (*it < 0) {
+                throw std::invalid_argument("Norms shouldn't be less than 0!");
+            }
+        }
+}
+
+template <typename InputT>
+NormalizeTransformer<InputT>::NormalizeTransformer(Archive &ar) :

```

Comment:
>NormalizeTransformer<InputT>::NormalizeTransformer(Archive &ar) [](start = 0, length = 63)

Add a unit test that throws an error when invalid version numbers are deserialized (see the other featurizers for an example of this).